frame/support/procedural/src/construct_runtime/expand/config.rs

```diff
@@ -47,6 +48,7 @@ pub fn expand_outer_config(
 			build_storage_calls.extend(expand_config_build_storage_call(scrate, runtime, decl, &field_name));
 			query_genesis_config_part_macros.push(quote! {
 				#path::__substrate_genesis_config_check::is_genesis_config_defined!(#pallet_name);
+				#path::__substrate_genesis_config_check::is_std_enabled_for_genesis!(#pallet_name, #path_str);
```

Comment:
Does this work, the "test" should only be run on std, when building the wasm we expect the pallet to have std not enabled. I think you should wrap it inside `#[cfg(feature = "std")]` item.
```suggestion
				#[cfg(feature = "std")]
				#path::__substrate_genesis_config_check::is_std_enabled_for_genesis!(#pallet_name, #path_str);
```